cymetric/graphs.py

```diff
@@ -46,20 +46,56 @@ def flow_graph(evaler, senders=(), receivers=(), commodities=(), nucs=(),
     df = df[group_start].groupby(group_end).sum()
     df.reset_index(inplace=True)
 
-    agents_ = evaler.eval('AgentEntry')['Prototype'].tolist()
+    agents_ = evaler.eval("AgentEntry")
 
-    dot = Digraph('G')
+    dot = Digraph("G", strict=True)
 
-    for agent in agents_:
-        dot.node(agent)
+    # start by constructing region subgraphs
+    regions = agents_[agents_["ParentId"] == -1]
```

Comment:
That's true, `agents_` is a `DataFrame` that also includes a column for `Kind` of agent type: `Region`, `Inst`, `Facility`. Presumably these are the _only_ three possible values for `Kind`? In that case, we could equivalently do:

```suggestion
    regions = agents_[agents_["Kind"] == "Region"]
```

Maybe that is more readable? I was working on the assumption that a region would never have a parent, and therefore it would have a `ParentId = -1`.

Reply:
I think that is better.  I agree that it's unlikely for any other type of agent to have no parent, but I don't think it's excluded.